CI/travis/setup_qemu_for_arm.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# Based on a test script from avsm/ocaml repo https://github.com/avsm/ocaml
+set -ex
+
+if [ $# -ne 1 ] ; then
+	echo Must include debian distribution ie wheezy, jessie, stretch, or buster
+	exit -1
+fi
+if [[ ! $1 =~ ^wheezy|jessie|stretch|buster$ ]] ; then
+	echo Must include debian distribution ie wheezy, jessie, stretch, or buster
+	exit -1
+fi
+
+CHROOT_DIR=/tmp/arm-chroot
+MIRROR=http://archive.raspbian.org/raspbian
+# wheezy = 7 (2013-05-04); jessie = 8 (2015-04-26); stretch = 9 (2017-06-17); buster = 10 (2019-03-12)
+VERSION=$1
+echo building for ${VERSION}
+CHROOT_ARCH=armhf
+
+# Host dependencies
+sudo apt-get update
+sudo apt-get install -qq -y qemu-user-static binfmt-support sbuild wget debian-archive-keyring ubuntu-keyring gnupg libudev1 libudev-dev liblzma-dev libglib2.0-dev libfdt-dev libpixman-1-dev zlib1g-dev
+# need a newer version of qemu :(
+QEMU_DIR=/tmp/qemu
+sudo mkdir -p ${QEMU_DIR}
+sudo chown ${USER} /tmp/qemu
+cd ${QEMU_DIR}
+wget https://download.qemu.org/qemu-3.1.0.tar.xz
+tar xf qemu-3.1.0.tar.xz
```

Comment:
curios: would the default QEMU work ?

Reply:
actually, there may be some good news here [no guarantees that it actually works though, but it would be worth taking a closer look] ;
you could use Travis-CI's integration with Debian Sid packages to get a new QEMU [3.1 or something]

QEMU in Sid seems to be 3.1 [now] : https://packages.debian.org/sid/qemu

What you typically would do, is:
```
addons:
  apt:
    sources:
    - debian-sid
    packages:
    - qemu
```

I "think" Travis-CI keeps a list of whitelisted packages; they don't seem to just allow anything from Sid. I don't remember exactly.

I did do this [Debian-Sid usage] a while back on some other project, and it's a mix of experiences ; some worked great, but if you mix-in other stuff [or APT sources] then using `debian-sid` can confuse other parts of the build.

Still [if possible] I would try it, mostly to reduce the build logic here, but especially to try to reduce the build-time.

Reply:
> curios: would the default QEMU work ?

No - it's too old, and gives errors during make package. it doesn't support some ioctls which cmake uses.

Reply:
> Still [if possible] I would try it, mostly to reduce the build logic here, but especially to try to reduce the build-time.

building qemu doesn't take the time (thats 2-3 min our of 20) - it's downloading the entire dist for ARM that takes the time...

Reply:
> QEMU in Sid seems to be 3.1 [now] : https://packages.debian.org/sid/qemu

We actually need qemu-user-static to run in the chroot; so it would be https://packages.debian.org/sid/qemu-user-static (which is still there).